test/unit/app/managers/test_DatasetManager.py

```diff
@@ -33,13 +33,13 @@ def set_up_managers(self):
 
     def test_create(self):
         self.log("should be able to create a new Dataset")
-        dataset1 = self.dataset_manager.create()
+        dataset1 = self.dataset_manager.create_for_unit_testing()
```

Comment:
I think we need to use `create_for_unit_testing()` only for the 3 tests that were failing: `test_create_public_dataset`, `test_create_private_dataset` and `test_serializers`.

```suggestion
        dataset1 = self.dataset_manager.create()
```

Reply:
I think we want to use the method designed to actually create datasets for all the tests though right? The other one just happens to accidentally work 😅. It shouldn't actually be used I imagine.

Reply:
I agree on the "It shouldn't actually be used", so what about https://github.com/galaxyproject/galaxy/commit/76aa052ed71fd2e6f454bd4f0c5fa1a527aa392e ?
If you like it, I can push to your branch.

Reply:
Sounds like a good fix - obviously better than leaving unit test fixutres in the main code base! I would say push it.